### PR TITLE
pr_parser, do not exit 1 when found merged PR

### PR DESCRIPTION
**Pull request by changev**
1. Add pr group valid status check
2. Recover the error handling when found "depends on invalid PR url"
3. fix bug: exit 1 when found merged PR
@panpan0000 @PengTian0

### CODE REVIEW

File: build-release-tools/application/pr_parser.py [hunk 1]
```diff
@@ -32,7 +32,20 @@ def __init__(self, change_url, target_branch, ghtoken):
         self.__pr_list = [self.__repo]
         self.__pr_connectivity_map = collections.defaultdict(dict)
         self.__gh = Github(ghtoken)
-        self.__all_prs = []         
+        # If the prs of this build is valid, 
+        # -1 haven't parse pr, 0 is valid, 1 invalid
+        # unmergeable pr will set this var to 1
+        self.__valid_pr_group = -1
```
Comment (panpan0000) [inline build-release-tools/application/pr_parser.py]: normally, _valid_xxx == 1 means it's valid.. but you choose the opposite ?

Comment (changev) [reply build-release-tools/application/pr_parser.py]: Thanks~~ I was taken to the wrong way by shell exit code.

**Comment by PengTian0** [comment]
:+1:

**Comment by PengTian0** [comment]
Please retest RackHD/12 after the pr merged

**Review by yyscamper — APPROVED** [review]

**Pull request merged**